pkg/operator/staticpod/file_utils.go

```diff
@@ -0,0 +1,37 @@
+package staticpod
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"io/ioutil"
+	"k8s.io/klog/v2"
+)
+
+func WriteFileAtomic(content []byte, filePerms os.FileMode, resource, contentDir, filename string) error {
+	tmpFile, err := writeTemporaryFile(content, filePerms, contentDir, filename)
+	if err != nil {
+		return err
+	}
+
+	klog.Infof("Renaming %s manifest %q to %q ...", resource, tmpFile, path.Join(contentDir, filename))
+	return os.Rename(tmpFile, path.Join(contentDir, filename))
+}
+
+func writeTemporaryFile(content []byte, filePerms os.FileMode, contentDir, filename string) (string, error) {
+	klog.Infof("Creating a temporary file for %q ...", path.Join(contentDir, filename))
```

Comment:
do we really need all this logging? Would streamline the func `WriteFileAtomic` to not output anything.